### PR TITLE
SNOW-1544013 Fetch events for app

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * N/A I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Fetches events from the account event table (See #1337). This does not implement filtering, timeboxing, or limiting output (these will come in a followup).

Regular output prints the messages as a log stream, including any ASCII control codes:
image

JSON output prints each message as individual JSON values:
image

This can then be piped to `jq` or equivalent:
image

### CODE REVIEW

File: src/snowflake/cli/plugins/nativeapp/manager.py [hunk 1]
```diff
@@ -89,7 +90,7 @@
 
 def generic_sql_error_handler(
     err: ProgrammingError, role: Optional[str] = None, warehouse: Optional[str] = None
-):
+) -> NoReturn:
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/manager.py:93]: nit: doesn't the rest of the code use -> None for this?

File: src/snowflake/cli/plugins/nativeapp/manager.py [hunk 2]
```diff
@@ -715,9 +716,25 @@ def get_validation_result(self, use_scratch_stage: bool):
...
     def get_events(self) -> list[dict]:
-        if self.account_event_table is None:
+        if not self.account_event_table:
             raise NoEventTableForAccount()
-        return []
+
+        app_name = (
+            unquote_identifier(self.app_name)
+            if is_valid_quoted_identifier(self.app_name)
+            else self.app_name.upper()
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/manager.py:725]: nit: unquote_identifier should already do pretty much all of this, including case normalization

File: src/snowflake/cli/plugins/nativeapp/manager.py (hunk 1 above)
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/plugins/nativeapp/manager.py:93]: `NoReturn` means it never returns (i.e. always raises an exception). This helps the type checker know that any code after a call to this function is unreachable

File: tests_integration/nativeapp/test_events.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
 
 @pytest.mark.integration
-def test_app_events(runner, temporary_working_directory):
+def test_app_events_no_event_table(runner, temporary_working_directory):
```
Comment (sfc-gh-fcampbell) [inline tests_integration/nativeapp/test_events.py:30]: I could add a test that creates a new event table, raw inserts events into it, and reads them back, but I'm worried about it being slow and/or flaky. Thoughts?

**Review by sfc-gh-bdufour — APPROVED** [review]

**Pull request merged**